byron-proxy/src/exec/Validator.hs

```diff
@@ -136,15 +148,25 @@ main = do
         stopCondition = const (pure Nothing)
         tracer = contramap (\txt -> ("", Info, txt)) trace
         client  = Client.chainSyncClient (clientFold tracer genesisConfig stopCondition cvs)
-        peer    = Client.chainSyncClientPeer client
-        codec   = ChainSync.codec epochSlots
-        action  = \socket -> runPeer nullTracer codec (socketAsChannel socket) peer
-        mkSocket = Socket.socket
-          (Socket.addrFamily addrInfo)
-          (Socket.addrSocketType addrInfo)
-          (Socket.addrProtocol addrInfo)
-    outcome <- bracket mkSocket Socket.close $ \socket -> do
-      _ <- Socket.connect socket (Socket.addrAddress addrInfo)
-      action socket
-    -- TODO print the outcome?
-    pure ()
+    connectTo
+      encodeTerm
+      decodeTerm
+      (initiatorVersions epochSlots client)
+      addrInfoLocal
+      addrInfoRemote
+
+-- Orphans, forced upon me because of the IO sim stuff.
+-- Required because we use ResourceT in the chain sync server.
+
+instance NonStandard.MonadThrow (ResourceT IO) where
+  throwM = Standard.throwM
```

Comment:
Do we have more of these?  We could gather them in a package `io-sim-instances`.

Reply:
They're orphans. What's considered best practice for orphans? I personally am not happy with having them at all. Is putting them into a package a good idea? @dcoutts suggested leaving them here in executables, and I agree that orphans way more acceptable in executables than in libraries.

Reply:
There's probably no good solution; my argument for a central place for orphans, is that it makes it easier to review & re-use them.  It's not in `io-sim-classes` (or `io-sim`) because we don't want  `resourcet` dependency there.

Reply:
Well, there is a good solution IMO: either

- don't put the non-standard `io-sim` `MonadThrow`, `MonadCatch`, or `MonadMask` class constraints on useful driver functions from `typed-protocols`. `runPeer`, for instance, should not have `MonadThrow`, it should instead return `Either unexpectedMessage a`.
- or if we insist on using those constraints, use the _standard_ ones! `ResourceT` already has `MonadThrow`, `MonadCatch`, and `MonadMask` instances, but for the _standard_ variants that _everyone else_ already uses.